primer3/thermoanalysis.pyx

```diff
@@ -219,41 +224,118 @@ cdef class ThermoResult:
         }
 
 
+def _try_get_dict_int(
+        argname: str,
+        key: Union[str, int],
+        dict_obj: Dict[str, int],
+) -> int:
+    '''Helper function to translate an string name to an integer arguments
+    using a dictionary:
+
+    Args:
+        argname:
+        key: integer value or string name key mapping to an integer
+        dict_obj: dictionary mapping the string to an int
+
+    Returns:
+        integer value for the key in the map
+
+    Raises:
+        KeyError: Key missing in the map
+        TypeError: invalid type for the key
+    '''
+    if isinstance(key, (int, long)):
+        return key
+    elif isinstance(key, str):
+        if key not in dict_obj:
+            raise ValueError(f'{argname}: {key} argument not in {dict_obj}')
+        return dict_obj[key]
+    else:
```

Comment:
I'm surprised that pylint doesn't bark about this -- probably don't need the `else` given that the other two branches both `return`